### ISSUE TITLE
Implement `eq1.subs(eq2)`

### ISSUE THREAD

**Issue by Davide-sd** [gutow/Algebra_with_Sympy#9]
Would it be possible to overwrite the `subs` method so that it can accept a single argument, specifically an object of type `Equation`? So, the left hand side represents the expression to substitute, the right hand side represents the new value.

**Comment by gutow** [comment]
I have thought about this some and think it should be done. I have not dug into what I would have to override in the subs module to make it work. I think it may be as simple as checking for equations and parsing them before calling the normal subs modules.

This is not completely straightforward in an external package as it requires overriding things defined in the `Expr` class. This code already does some pretty substantial tweaking of the namespace to extend the FunctionClass to work with equations. It would all be much easier if this could be incorporated directly into sympy, but we have not made progress on making the decisions necessary to accept https://github.com/sympy/SymPEPs/pull/1, which would roll them together.

It is on my radar, but relatively low priority compared to getting more tests built so this package can get to the stable stage and continuing to try to get a positive consensus on embedding this code in sympy.

If you or someone else can get this working, I would love to get a pull-request.

**Comment by Davide-sd** [comment]
> If you or someone else can get this working, I would love to get a pull-request.

I might give it a try next week! :)

**Comment by gutow** [comment]
> I might give it a try next week! :)

Thank you.

**Comment by Davide-sd** [comment]
I gave it a quick try, and it seems that the following could work:

```
    def subs(self, *args, **kwargs):
        new_args = args
        if (len(args) == 1) and isinstance(args[0], self.func):
            new_args = [{args[0].args[0]: args[0].args[1]}]
        return super().subs(*new_args, **kwargs)
```

While I'd really love the possibility of simply writing `eq1.subs(eq2)`, there is a conceptual question that we should discuss first.

Currently, `subs` accepts these type of arguments (no matter the type of object we are modifying (expression, matrix, vector, ...)):
* `subs(old, new)`
* `subs([(old1, new1), (old2, new2), ...])`
* `subs({old1: new1, ...})`

Do we really want to modify the signature to implement this new behavior? (I vote for yes :) )

Having played with the module on a few examples, I noticed that often I need to substitute multiple equations into a target equation. An alternative would be to create a function `subs_dict(*eqns)` that, given a list of equations it creates a substitution dictionary where each key represents the LHS and the value represents the RHS. Result: `eq1.subs(subs_dict(eq2, eq3, ...))`

Maybe we can implement both approaches...
